core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/keymanager/delete/DeleteKeystoresHandler.java

```diff
@@ -0,0 +1,198 @@
+        parsedBody.getPubkeys().stream()
+            .map(IdentifierUtils::normaliseIdentifier)
+            .collect(Collectors.toList());
+
+    final List<String> keysToExport = new ArrayList<>();
+    // attempt to delete keys one by one, and return results with statuses
+    final List<DeleteKeystoreResult> results = processKeysToDelete(pubkeysToDelete, keysToExport);
+    // export slashing data for keys that were either 'deleted' or 'not_active'
+    final String slashingProtectionExport = exportSlashingProtectionData(results, keysToExport);
+
+    try {
+      context
+          .response()
+          .putHeader(CONTENT_TYPE, JSON_UTF_8)
+          .setStatusCode(SUCCESS)
+          .end(
+              objectMapper.writeValueAsString(
+                  new DeleteKeystoresResponse(results, slashingProtectionExport)));
+    } catch (JsonProcessingException e) {
+      context.fail(SERVER_ERROR, e);
+    }
+  }
+
+  private List<DeleteKeystoreResult> processKeysToDelete(
+      List<String> pubkeysToDelete, List<String> keysToExport) {
+    final List<DeleteKeystoreResult> results = new ArrayList<>();
+    // process each incoming key individually
+    for (String pubkey : pubkeysToDelete) {
+      try {
+        final Optional<ArtifactSigner> signer = signerProvider.getSigner(pubkey);
+        final Optional<List<Path>> keystoreConfigFiles =
+            keystoreFileManager.findKeystoreConfigFiles(pubkey);
+
+        // check that key is active
+        if (signer.isEmpty()) {
+          // if not active, check if we ever had this key registered in the slashing DB
+          final boolean wasRegistered =
+              slashingProtection
+                  .map(protection -> protection.isRegisteredValidator(Bytes.fromHexString(pubkey)))
+                  .orElse(false);
+
+          // if it was registered previously, return not_active and add to list of keys to export,
+          // otherwise not_found
+          if (wasRegistered) {
+            keysToExport.add(pubkey);
+            results.add(new DeleteKeystoreResult(DeleteKeystoreStatus.NOT_ACTIVE, ""));
+          } else {
+            results.add(new DeleteKeystoreResult(DeleteKeystoreStatus.NOT_FOUND, ""));
+          }
+          continue;
+        }
+
+        // Check that key is read only, if so return an error status
+        if (signer.get() instanceof BlsArtifactSigner
+            && ((BlsArtifactSigner) signer.get()).isReadOnlyKey()) {
+          results.add(
+              new DeleteKeystoreResult(
+                  DeleteKeystoreStatus.ERROR, "Unable to delete readonly key: " + pubkey));
+          continue;
+        }
+
+        // Remove active key from memory first, will stop any further signing with this key
+        signerProvider.removeSigner(pubkey).get();
+        // Then, delete the corresponding keystore file
+        if (keystoreConfigFiles.isPresent()) {
+          for (final Path path : keystoreConfigFiles.get()) {
```

Comment:
Seems like we don't need to the key config files we just need to be able to delete them. Perhaps change the KeystoreFileManager api to have a delete method instead of a findKeystoreConfigFiles method.

Reply:
done